src/tribol/mesh/CouplingScheme.hpp

```diff
@@ -82,6 +86,58 @@ class InterfacePairs;
 class CouplingScheme
 {
 public:
+  template <typename T,
+            typename PARAM,
+            typename MESH,
+            typename CP,
+            typename CP2D,
+            typename CP3D>
+  class ViewerBase
+  {
+  public:
+    ViewerBase( T& cs );
+
+    TRIBOL_HOST_DEVICE int spatialDimension() const { return m_mesh1.spatialDimension(); }
+
+    TRIBOL_HOST_DEVICE const MeshData::Viewer& getMesh1() const { return m_mesh1; }
+
+    TRIBOL_HOST_DEVICE const MeshData::Viewer& getMesh2() const { return m_mesh2; }
+
+    TRIBOL_HOST_DEVICE const EnforcementOptions& getEnforcementOptions() const { return m_enforcement_options; }
+
+    TRIBOL_HOST_DEVICE CP& getContactPlane( IndexT id ) const;
+
+    TRIBOL_HOST_DEVICE RealT getTimestepScale() const { return m_parameters.timestep_scale; }
+
+    /*!
+    * Get the gap tolerance that determines in contact face-pairs for each
+    * supported interface method
+    *
+    * \return the gap tolerance based on the method
+    */
+    TRIBOL_HOST_DEVICE RealT getCommonPlaneGapTol( int fid1, int fid2 ) const;
```

Comment:
Thought it might be cleaner if we got rid of the switchyard in the function since common plane is the only method that uses a gap tolerance.  I can re-introduce if preferred.  More generally, method specific calcs should probably be moved out of the coupling scheme to avoid switch statements like these.